dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java

```diff
@@ -312,22 +366,42 @@ void readBlock() throws HpccFileException
             }
             catch (IOException e)
             {
-                throw new HpccFileException(e.getMessage());
+                prefetchException = new HpccFileException(e.getMessage());
             }
             return;
         }
 
         if (len < 4)
         {
-            throw new HpccFileException("Early data termination, no handle");
+            prefetchException = new HpccFileException("Early data termination, no handle");
+            try
+            {
+                close();
+            }
+            catch(Exception e){}
+            return;
         }
 
         try
         {
             this.handle = dis.readInt();
             if (this.handle == 0)
             {
-                len = retryWithCursor();
+                try
+                {
+                    len = retryWithCursor();
```

Comment:
I can't tell if retryWithCursor generates a trace log entry, but if it doesn't let's make sure we notify we're trying to handle a restart due to the received 0 handle